templates/docred/templates.yaml

```diff
@@ -0,0 +1,51 @@
+dataset: docred
+templates:
+  02af700c-e9e9-4a84-b75c-5fb29a5b7993: !Template
+    id: 02af700c-e9e9-4a84-b75c-5fb29a5b7993
+    jinja: "Read the following text and answer the questions.\n\nText:\n{% for sent\
+      \ in sents -%}\n{{ sent | join(\" \") }}\n{% endfor %} \n\nQuestion:\n\nFrom\
+      \ above text, find people, locations, organizations, time, numbers, and miscellaneous.\n\
+      |||\nPeople:\n{% for ners in vertexSet -%}\n{% for ner in ners if ner['type']\
+      \ == 'PER' -%}\n{{ner['name']}}{{\", \"}}\n{%- endfor %}\n{%- endfor %}\n\n\
+      Locations:\n{% for ners in vertexSet -%}\n{% for ner in ners if ner['type']\
+      \ == 'LOC' -%}\n{{ner['name']}}{{\", \"}}\n{%- endfor %}\n{%- endfor %}\n\n\
+      Organizations:\n{% for ners in vertexSet -%}\n{% for ner in ners if ner['type']\
+      \ == 'ORG' -%}\n{{ner['name']}}{{\", \"}}\n{%- endfor %}\n{%- endfor %}\n\n\
+      Time:\n{% for ners in vertexSet -%}\n{% for ner in ners if ner['type'] == 'TIME'\
+      \ -%}\n{{ner['name']}}{{\", \"}}\n{%- endfor %}\n{%- endfor %}\n\nNumbers:\n\
+      {% for ners in vertexSet -%}\n{% for ner in ners if ner['type'] == 'NUM' -%}\n\
+      {{ner['name']}}{{\", \"}}\n{%- endfor %}\n{%- endfor %}\n\nMiscellaneous:\n\
+      {% for ners in vertexSet -%}\n{% for ner in ners if ner['type'] == 'MISC' -%}\n\
+      {{ner['name']}}{{\", \"}}\n{%- endfor %}\n{%- endfor %}"
+    name: type-to-entity
+    reference: ''
+    task_template: false
+  3ab9cfc9-3ba3-41dd-959c-60182def11af: !Template
+    id: 3ab9cfc9-3ba3-41dd-959c-60182def11af
+    jinja: "Read the following text and answer the questions.\n\nText:\n{% for sent\
+      \ in sents -%}\n{{ sent | join(\" \") }}\n{% endfor %} \n\nQuestion:\n\nAssign\
+      \ an entity type to the following entities. The choices are PER (Person), LOC\
+      \ (Location), ORG (Organization), TIME (Time), NUM (Number), and MISC (Miscellaneous).\n\
+      \nExample: \n{% for ners in vertexSet[:1] %}\n{% for ner in ners %}\n{{ner['name']}}:\
+      \ {{ner['type']}}\n{% endfor %}\n{% endfor %}\n\nNow do the same with below:\n\
```

Comment:
This ends up being a bit weird because entities are duplicated in the list. Can you deduplicate?

Reply:
The duplication happens because multiple of the same entities can be detected throughout the text, and that's what the labels of the dataset give us. I don't think it's trivial to deduplicate them but if you have a better idea, please let me know.

Reply:
I think you should be able to use this filter: https://jinja.palletsprojects.com/en/3.0.x/templates/#jinja-filters.unique